borg/repository.py

```diff
@@ -83,6 +85,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is not None:
             self.rollback()
+            if hasattr(exc_val, 'errno'):
+                if exc_val.errno in [errno.EIO, errno.ENXIO]:
+                    return
         self.close()
```

Comment:
The context manager's `__exit__` is "on the outside", since the context manager is used like this in borg.archiver:

```
with Repository:
  do_create/delete/...:
     do_something_with_data
        do_something_with_repo
```

Therefore the exceptions that "travel by" this method could've been raised pretty much anywhere.